mypy/semanal_enum.py

```diff
@@ -106,10 +106,33 @@ def parse_enum_call_args(self, call: CallExpr,
         args = call.args
         if len(args) < 2:
             return self.fail_enum_call_arg("Too few arguments for %s()" % class_name, call)
-        if len(args) > 2:
-            return self.fail_enum_call_arg("Too many arguments for %s()" % class_name, call)
-        if call.arg_kinds != [ARG_POS, ARG_POS]:
-            return self.fail_enum_call_arg("Unexpected arguments to %s()" % class_name, call)
+        elif len(args) == 2:
+            if call.arg_kinds != [ARG_POS, ARG_POS]:
+                return self.fail_enum_call_arg("Unexpected arguments to %s()" % class_name, call)
+        elif len(args) > 2:
+            if len(args) == 3:
```

Comment:
This all seems correct but could you please refactor it to be a little more concise and remove some of the repetition? Instead of enumerating all possible cases, I'd suggest checking that there are between 2 and 6 arguments, that the first two are ARG_POS, and that the rest are ARG_NAMED_OPT.

Thanks!

Reply:
Alright :)